### PR TITLE
[NSIS Reference] Modernize + Add Category Filter

### PR DESCRIPTION
**Pull request by xmok**
## Description

Added a Dropdown so we can filter the documentation items by category. I also modernized the extension by updating deps, adding TS, adding CHANGELOG, README, keywords, metadata images etc.

## Screencast

https://github.com/user-attachments/assets/2c77f4a6-96da-4328-90b2-79106aa301e9

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/nsis-reference/src/index.tsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
-import { ActionPanel, Detail, Icon, List, PushAction } from "@raycast/api";
+import { ActionPanel, Detail, Icon, List, Action } from "@raycast/api";
+// @ts-expect-error Cannot find module '@nsis/docs' or its corresponding type declarations.ts (2307)
```
Comment (idleberg) [inline extensions/nsis-reference/src/index.tsx:2]: I'd like to address this in `@nsis/docs`, so hopefully we can get rid off this comment

File: extensions/nsis-reference/src/index.tsx [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
 import { Callbacks, Commands, Includes, Plugins, Variables } from "@nsis/docs";
```
Comment (idleberg) [inline extensions/nsis-reference/src/index.tsx:3]: I will fix this later today in `@nsis/docs`, so we can get rid off this comment

**Comment by idleberg** [comment]
@raycastbot The changes look good to me!

File: extensions/nsis-reference/src/index.tsx (hunk 2 above)
Comment (xmok) [reply extensions/nsis-reference/src/index.tsx:3]: That was super fast of you!
Unfortunately, now I'm getting a new error:
`Cannot find module '@nsis/docs' or its corresponding type declarations.
  There are types at '[PATH]/raycast-extensions/extensions/nsis-reference/node_modules/@nsis/docs/dist/index.d.ts', but this result could not be resolved under your current 'moduleResolution' setting. Consider updating to 'node16', 'nodenext', or 'bundler'.ts (2307)`.
  
These settings are the Raycast set defaults but I can make the error go away by adding the path to tsconfig.json like so:

```json{
  "$schema": "https://json.schemastore.org/tsconfig",
  "include": ["src/**/*", "raycast-env.d.ts"],
  "compilerOptions": {
    ....
    "paths": {
      "@nsis/docs": [
        "./node_modules/@nsis/docs/dist/index.d.ts"
      ]
    }
  }
}`

Or we can let the previous comment remain. What would you like to do?

**Comment by pernielsentikaer** [comment]
It looks like this one isn't ready yet. Is that correct

**Comment by xmok** [comment]
Hey we can get this merged as the functionality is good to go 👍.
We can remove the comment in a future version since it's not a blocker.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**